WordPress/gutenberg
# Add automated core blocks documentation

## Description

Adds a script to generate core block documentation reading all the blocks in the `packages/block-library` and using their `block.json` properties to generate the list.

**Done:** 
- [x] Improve supports and attributes nesting values.
- [x] Hook it up to `docs:build` command


## How has this been tested?

- Run script using: `node bin/api-docs/gen-block-lib-list.js`
- Confirm results write to: `docs/reference-guides/core-blocks.md`

## Screenshots 

- Link to [rendered view of new docs](https://github.com/WordPress/gutenberg/blob/docs/gen-blocks-list/docs/reference-guides/core-blocks.md).

## Types of changes

- New script in bin/api-docs/gen-block-lib-list.js to generate core-blocks document
- Lists all core blocks with description, category, supports, attributes
